core/src/test/java/com/exadel/aem/backpack/core/services/impl/PackageServiceImplTest.java

```diff
@@ -254,22 +281,210 @@ public void shouldNotCreatePackageIfSuchPackageAlreadyExist() throws RepositoryE
+            PackageInfo aPackage = packageService.createPackage(resourceResolver, packageModel);
 
             assertEquals(PackageStatus.ERROR, aPackage.getPackageStatus());
             assertEquals("ERROR: Package does not contain any valid filters.", aPackage.getLog().get(0));
         }
 
-        private void initBasePackageInfo(final CreatePackageModel model, final List<String> strings) {
-            model.setPaths(strings);
+        private void initBasePackageInfo(final PackageModel model, final List<String> strings, final boolean excludeChildren) {
+            model.setPaths(strings.stream().map(s -> new PathModel(s, excludeChildren)).collect(Collectors.toList()));
             model.setPackageName(TEST_PACKAGE);
             model.setThumbnailPath(THUMBNAIL);
             model.setVersion(PACKAGE_VERSION);
         }
     }
 
+    public static class EditPackage extends Base {
+        private static final String TEST_GROUP_2 = "testGroup2";
+        private static final String TEST_PACKAGE_2 = "testPackage2";
+        private JcrPackage aPackage;
+        private List<PathModel> expectedInitialFiltersModels = Collections.singletonList(new PathModel(PAGE_1, false));
+
+        private PackageInfo packageInfo;
+
+        @Before
+        public void before() throws IOException, RepositoryException {
+            packageInfo = new PackageInfo();
+            packageInfo.setGroupName(TEST_GROUP);
+            packageInfo.setPackageName(TEST_PACKAGE);
+            packageInfo.setVersion(PACKAGE_VERSION);
+            packageInfo.setReferencedResources(referencedResources);
+            packageInfo.setPaths(Collections.singletonList(PAGE_1));
+            packageInfo.setPackagePath(PACKAGE_PATH);
+            aPackage = createPackage(packageInfo, Collections.singletonList(new PathModel(PAGE_1, false)), new DefaultWorkspaceFilter());
+            resourceResolver = context.resourceResolver();
+        }
+
+
+        @Test
+        public void shouldEditPackage() throws RepositoryException {
+            PackageModel packageModel = new PackageModel();
+            HashSet<AssetReferencedItem> assetReferenceItems = new HashSet<>();
+            assetReferenceItems.add(new AssetReferencedItem(PICTURE_3, IMAGE_JPEG));
+            when(referenceServiceMock.getAssetReferences(any(ResourceResolver.class), any(String.class))).thenReturn(assetReferenceItems);
+
+            Map<String, List<String>> modifiedReferencedResources = new HashMap<>();
+            modifiedReferencedResources.put(IMAGE_JPEG, Collections.singletonList(PICTURE_3));
+
+            initBasePackageModel(packageModel, Collections.singletonList(PAGE_2), false);
+
+
+            PackageInfo aPackage = packageService.editPackage(resourceResolver, packageModel);
+
+
+            assertEquals(MODIFIED, aPackage.getPackageStatus());
+            assertNotNull("testPackage2-2.zip", aPackage.getPackageNodeName());
```

Comment:
These literals should be probably made constants, just because most of the others are, and they are repeating as well

Reply:
Fixed